### ISSUE TITLE
Stone Bricks replaced by wooden cranks

### ISSUE THREAD

**Issue by Wolf68k** [AppliedEnergistics/Applied-Energistics-2#1612]
Updated from stable3 to stable9. Created a new world for pack testing and after a bit of exploring the game crashed. 
Snip-it of said crash...
Snip part 1:
java.lang.StackOverflowError: Exception while updating neighbours
	at java.security.AccessController.doPrivileged(Native Method)
	at java.lang.ClassLoader.checkPackageAccess(Unknown Source)
	at net.minecraft.crash.CrashReportCategory.func_147153_a(CrashReportCategory.java:232)
	at net.minecraft.world.World.func_147460_e(World.java:729)
	at net.minecraft.world.World.func_147459_d(World.java:648)
	at net.minecraft.world.World.func_147444_c(World.java:613)
	at net.minecraft.world.World.markAndNotifyBlock(World.java:488)
	at net.minecraft.world.World.func_147465_d(World.java:466)
	at net.minecraft.world.World.func_147480_a(World.java:594)
	at appeng.block.grindstone.BlockCrank.dropCrank(BlockCrank.java:86)
	at appeng.block.grindstone.BlockCrank.func_149695_a(BlockCrank.java:144)

Snip part 2:
 -- Block being updated --
Details:
	Source block type: ID #255 (tile.appliedenergistics2.BlockCrank // appeng.block.grindstone.BlockCrank)
	Block type: ID #255 (tile.appliedenergistics2.BlockCrank // appeng.block.grindstone.BlockCrank)
	Block data value: 0 / 0x0 / 0b0000
	Block location: World: (231,77,-1670), Chunk: (at 7,4,10 in 14,-105; contains blocks 224,0,-1680 to 239,255,-1665), Region: (0,-4; contains chunks 0,-128 to 31,-97, blocks 0,0,-2048 to 511,255,-1537)

Both parts repeat several times. When I go to location I find a village that should have a Stone Bricks wall, instead it's wooden cranks on the ground. In another test I found a Railcraft building and the same thing happened to the roof only this time I was able to stand/walk on invisible blocks, however no cranks to be found.

Complete crash report: http://pastebin.com/vgjuZ7GU

Screen shots:
http://i.imgur.com/Cqkb6Mt.jpg
http://i.imgur.com/YV5BJHS.jpg
http://i.imgur.com/JV8khqO.jpg
http://i.imgur.com/pIYfREO.jpg

**Comment by thatsIch** [comment]
I think something messed up with the ID allocation, imo this is unrelated to AE2, this could happen to any mod.

Indication is, that a block with the ID <= 255 is replaced (no mod block has such a block ID, but if you inject it directly into the WorldGeneration like Biomes'O'Plenty does)

The crash just states, that you update too many blocks at once.


Are you sure, you just updated AE2 only?

**Comment by thatsIch** [comment]
Can you try to reproduce this with AE2 only / see with which minimal mods you reproduce this?

**Comment by yueh** [comment]
Can you also exclude that project red is the issue? There are a couple of reports that it messes with the stone bricks recipe, so messing with the block itself does not sound that unlikely.

**Comment by Wolf68k** [comment]
I didn't update only AE2. I updated a lot of mods.
I checked in a previous instance, one that is public and has no major issues, and the wooden crank is ID 255

I tried downgrading to the previous version of ProjectRed, the same one that is being used in public version of the pack. It seems ProjectRed is the issue.

Sorry to have wasted your time but then thank you for the ProjectRed insight. I wouldn't have thought of that.
